uniffi_bindgen/src/bindings/kotlin/gen_kotlin/mod.rs

```diff
@@ -510,6 +511,29 @@ mod filters {
         Ok(as_ct.as_codetype().literal(literal, ci))
     }
 
+    // Get the idiomatic Kotlin rendering of an individual enum variant's discriminant
+    pub fn variant_discr_literal(e: &Enum, index: &usize) -> Result<String, askama::Error> {
+        let literal = e.variant_discr(*index).expect("invalid index");
+        match literal {
+            // Kotlin doesn't convert between signed and unsigned by default
+            // so we'll need to make sure we define the type as appropriately
+            LiteralMetadata::UInt(v, _, _) => {
+                if let Some(t) = e.variant_discr_type() {
+                    match t {
+                        Type::Int8 | Type::Int16 | Type::Int32 | Type::Int64 => Ok(v.to_string()),
```

Comment:
@mhammond curious your thoughts on this, I know it's a bit "hacky" until we get full support for signed integers but prevents current users from breaking when using the proc-macro.

Reply:
I don't quite understand this - it looks like this is going to write the signed value to the generated kotlin as unsigned - ie, that in the Kotlin test, when we add the test for `AnimalSignedInt` we'd find it written as though it was unsigned. If that's true, that's not really what I would expect. If it's false I don't quite understand what it would be doing here.